src/key_value_store/mod.rs

```diff
@@ -0,0 +1,52 @@
+use super::event::Event;
+use core::cell::Cell;
+
+pub trait KeyValueStore<'a> {
+    fn read<T: SafelyDeserializable + Sized>(&self, key: Key) -> T;
+    fn write<T: Sized>(&self, key: Key, value: &T);
```

Comment:
These generic functions mean that we can't use `KeyValueStore` to refer to abstract objects. This will have to be reworked to inject the `SafelyDeserializable` during read and do... something during write.